src/setup/node.rs

```diff
@@ -171,16 +141,72 @@ impl Drop for Node {
     }
 }
 
+/// Convenience struct to better control configuration/build/start for [Node]
+pub struct NodeBuilder {
+    config: NodeConfig,
+    meta: NodeMetaData,
+}
+
+impl NodeBuilder {
+    /// Sets up minimal configuration for the node.
+    pub fn new(path: PathBuf, ip_addr: IpAddr) -> Result<Self> {
+        let config = NodeConfig::new(path, ip_addr);
+        let meta = NodeMetaData::new(config.path.clone())?;
+        Ok(Self { config, meta })
+    }
+
+    /// Sets initial peers for the node.
+    pub fn initial_peers(mut self, addrs: Vec<SocketAddr>) -> Self {
+        self.config.initial_peers = addrs.into_iter().collect();
+        self
+    }
+
+    /// Sets validator token to be placed in rippled.cfg.
+    /// This will make the node to be run in validator mode.
```

Comment:
Does this mean the node _is_ a validator or does it mean it enables connections to validators?

Reply:
It turns the node into a validator. As per: https://xrpl.org/run-rippled-as-a-validator.html

Reply:
In that case I'd suggest:
```suggestion
    /// This will configure the node to run as a validator. 
```

Reply:
Done